src/thunderbots/software/backend/robot_status.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+// This struct contains various robot diagnostics (e.g. voltages, link quality, etc.)
+// and messages related to robot and dongle condition.
+struct RobotStatus
+{
+    // The robot number.
+    int32_t robot;
+
+    // Messages related to the robot that are to be displayed in the visualizer
+    // (usually errors that need attention)
+    std::vector<std::string> robot_messages;
+
+    // Messages that are related to the dongle itself
+    // (usually errors that need attention)
+    std::vector<std::string> dongle_messages;
+
+    // The maximum possible kick speed, in m/s.
+    double kick_speed_max;
+
+    // The kick resolution for HScale.
+    double kick_speed_resolution;
+
+    // The maximum possible chip distance, in m.
+    double chip_distance_max;
+
+    // The chip resolution for HScale.
+    double chip_distance_resolution;
+
+    // The maximum power level understood by the direct_dribbler function.
+    uint32_t direct_dribbler_max;
+
+    // Whether or not the robot is currently responding to radio communication.
+    bool alive;
+
+    // Whether the robot is in direct mode.
+    bool direct_control;
+
+    // Whether or not the ball is interrupting the robot’s laser beam.
+    bool ball_in_beam;
+
+    // Whether or not the robot’s capacitor is charged enough to kick the ball.
+    bool capacitor_charged;
```

Comment:
I might rename this to something more specific. Could easily be mistaken for the capacitor being at full charge

Reply:
Technically out-of-scope (copied over from ROS message), but an easy fix, changed to `capacitor_charged_enough_to_kick_ball`